src/compiler/conversion/mod.rs

```diff
@@ -159,6 +159,11 @@ impl Conversion {
                 let parsed = s
                     .parse::<f64>()
                     .with_context(|_| FloatParseSnafu { s: s.clone() })?;
+                if parsed.is_infinite() {
```

Comment:
This is quite nuanced, but I think the least surprising behavior is to use https://doc.rust-lang.org/std/primitive.f64.html#method.is_normal here.